### ISSUE TITLE
check_syntax: false positive for assignment in nested DDIC structures

### ISSUE THREAD

**Issue by BlackmanCC** [abaplint/abaplint#2318]
I have a nested DDIC structure:

```
define type zeide_nad {
  name  : include zeide_name;
  address : include zeide_address;
  comid : zeidecomid;

}
```
When I want to fill it (re_snad is of type zeide_nad) with 

`    re_snad-name = read_nad_name_in( im_snad ).`

I get a false positive in

> Details der Analyse
> 
> Not a structure, type unknown, target (check_syntax)

**Comment by larshp** [comment]
thanks, can you try reproducing it in a public github repository?

**Comment by BlackmanCC** [comment]
difficult. I do not have a public repo with abaplint running..

**Comment by larshp** [comment]
just make some dummy objects, https://github.com/marketplace/abaplint is one click setup, and free for public repos

need a bit more context to find the bug

**Comment by larshp** [comment]
also try opening the file in vscode, and hover the variable to check what it says,

![image](https://user-images.githubusercontent.com/5888506/147677092-cb21f8e7-4769-4844-ba3e-c31393773a11.png)

**Comment by BlackmanCC** [comment]
Please check repo [abaplint_dummy](https://github.com/BlackmanCC/abaplint_dummy)

I hope this will help.

**Comment by larshp** [comment]
see comments in https://github.com/BlackmanCC/abaplint_dummy/pull/1

**Comment by BlackmanCC** [comment]
the two structures are now part of the package/repo, but error message is still there

**Comment by larshp** [comment]
any new messages if you enable rule `unknown_types` ?

**Comment by BlackmanCC** [comment]
Please check the example in my repo again. I removed all stuff not needed for the example and renamed variables. 
With both settings for "unknown types" we now have the same finding. 

> Component "substruc" not found in structure (check_syntax)

**Comment by larshp** [comment]
confirmed, FIELD2 and FIELD3 not on the right level, GROUPNAME not taken into account

![image](https://user-images.githubusercontent.com/5888506/147734254-3d91e6ab-f41e-488f-b1db-e4804477c24b.png)

**Comment by larshp** [comment]
@BlackmanCC try again, 2.83.13 published

**Comment by BlackmanCC** [comment]
it works now. THX!
